spatialos/UnrealGDK
# Make GDK Assembly packaging and build executable support Client targets

#### Description
Added support for the main build target having `TargetType.Client` (`<ProjectName>.Target.cs`). This target will be automatically built with arguments `-client -noserver` passed to UAT when building from the editor. If you use the GDK build script or executable manually, you need to pass `-client -noserver` when building this target (for example, `BuildWorker.bat GDKShooter Win64 Development GDKShooter.uproject -client -noserver`).

#### Tests
How can this be verified by QA?
- Without changing the main build target (`GDKShooter.Target.cs`), build Example project using `BuildWorker.bat`:
`BuildWorker.bat GDKShooter Win64 Development GDKShooter.uproject`
`BuildWorker.bat GDKShooterSimulatedPlayer Win64 Development GDKShooter.uproject`
- Build Example project from the editor from the editor cloud deployment window

In `GDKShooter.Target.cs`, change the target type to `TargetType.Client`, like so:
```
    public GDKShooterTarget(TargetInfo Target) : base(Target)
    {
        Type = TargetType.Client;
```
- Build Example project using `BuildWorker.bat`, additionally passing `-client -noserver`:
`BuildWorker.bat GDKShooter Win64 Development GDKShooter.uproject -client -noserver`
`BuildWorker.bat GDKShooterSimulatedPlayer Win64 Development GDKShooter.uproject -client -noserver`
- Build Example project from the editor from the editor cloud deployment window

All build are expected to succeed and the cloud deployments successfully started.

#### Primary reviewers
@mironec @joshuahuburn
